moirae/models/ecm/__init__.py

```diff
@@ -145,12 +145,13 @@ def calculate_terminal_voltage(
         Vt = asoh.ocv(soc=transient_state.soc, temp=new_inputs.temperature)
 
         # Add I*R drop ('DCIR')
-        Vt += new_inputs.current * asoh.r0.get_value(soc=transient_state.soc,
-                                                     temp=new_inputs.temperature)
+        IR_drop = new_inputs.current * asoh.r0.get_value(soc=transient_state.soc,
+                                                         temp=new_inputs.temperature)
+        Vt = Vt + IR_drop
 
         # Check series capacitance
         if transient_state.q0 is not None:
-            Vt += transient_state.q0.copy() / asoh.c0.get_value(soc=transient_state.soc.copy())
+            Vt = Vt + transient_state.q0.copy() / asoh.c0.get_value(soc=transient_state.soc.copy())
```

Comment:
Minor performance optimization: I believe we can get rid of the `.copy()` on these, especially for `.q0`. Their values are not being altered so we need not protect them using copy.